addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,719 @@
+        FROM account_move am
+        WHERE am.old_invoice_id = aipr.invoice_id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_analytic_tag_account_move_line_rel (
+            account_move_line_id, account_analytic_tag_id)
+        SELECT aml.id, aatailr.account_analytic_tag_id
+        FROM account_analytic_tag_account_invoice_line_rel aatailr
+        JOIN account_move_line aml
+        ON aml.old_invoice_line_id = aatailr.account_invoice_line_id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_analytic_tag_account_move_line_rel (
+            account_move_line_id, account_analytic_tag_id)
+        SELECT aml.id, aataitr.account_analytic_tag_id
+        FROM account_analytic_tag_account_invoice_tax_rel aataitr
+        JOIN account_move_line aml
+        ON aml.old_invoice_tax_id = aataitr.account_invoice_tax_id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move_line_account_tax_rel (
+            account_move_line_id, account_tax_id)
+        SELECT aml.id, ailt.tax_id
+        FROM account_invoice_line_tax ailt
+        JOIN account_move_line aml
+        ON aml.old_invoice_line_id = ailt.invoice_line_id
+        """,
+    )
+
+
+def migration_voucher_moves(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move am
+        SET (message_main_attachment_id, name, date, ref, narration, type,
+        journal_id, company_id, currency_id, partner_id, amount_tax,
+        amount_total, state, invoice_payment_state, invoice_date,
+        invoice_date_due, invoice_payment_ref, create_uid, create_date,
+        write_uid, write_date) = (
+        av.message_main_attachment_id, COALESCE(av.number, am.name),
+        COALESCE(av.date, am.date), av.reference, av.narration,
+        CASE WHEN av.voucher_type = 'purchase' THEN 'in_receipt'
+        ELSE 'out_receipt' END, av.journal_id, av.company_id,
+        av.currency_id, av.partner_id, av.tax_amount, av.amount,
+        CASE WHEN av.state = 'proforma' THEN 'posted' ELSE av.state END,
+        CASE WHEN av.state = 'cancel' THEN 'not_paid' ELSE 'paid' END,
+        av.account_date, av.date_due, av.name, av.create_uid, av.create_date,
+        av.write_uid, av.write_date)
+        FROM account_voucher av
+        WHERE am.old_voucher_id = av.id AND av.state not in (
+            'draft', 'cancel', 'proforma')
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move (message_main_attachment_id, name, date, ref,
+        narration, type, journal_id, company_id, currency_id, partner_id,
+        amount_tax, amount_total, state, invoice_payment_state, invoice_date,
+        invoice_date_due, invoice_payment_ref, old_voucher_id,
+        create_uid, create_date, write_uid, write_date)
+        SELECT message_main_attachment_id, number, date, reference, narration,
```

Comment:
```suggestion
        SELECT message_main_attachment_id, COAELESCE(number, '/'), date, reference, narration,
```

Sometimes draft voucher have no number

Reply:
> Sometimes draft voucher have no number

After the migration, we still had invoices with name='' (empty value not null).